File: 프로그래머스/lv1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    int num1[5] = {1, 2, 3, 4, 5};
    int num2[8] = {2, 1, 2, 3, 2, 4, 2, 5};
    int num3[10] = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    vector<int> total(3, 0);
    
    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == num1[i % 5]) total[0]++;
        if (answers[i] == num2[i % 8]) total[1]++;
        if (answers[i] == num3[i % 10]) total[2]++;
    }
    
    int max_val = *max_element(total.begin(), total.end());
    vector<int> answer;
    
    for (int i = 0; i < 3; i++) {
        if (total[i] == max_val) answer.push_back(i + 1);
    }
    
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    num1 = [1, 2, 3, 4, 5]
    num2 = [2, 1, 2, 3, 2, 4, 2, 5]
    num3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    total = [0] * 3
    
    for i, ans in enumerate(answers):
        if ans == num1[i % len(num1)]:
            total[0] += 1
        if ans == num2[i % len(num2)]:
            total[1] += 1
        if ans == num3[i % len(num3)]:
            total[2] += 1
    
    max_val = max(total)
    
    return [i + 1 for i, v in enumerate(total) if v == max_val]